buildSrc/subprojects/integration-testing/src/main/kotlin/gradlebuild/distribution-testing.gradle.kts

```diff
@@ -16,15 +16,25 @@
 package gradlebuild
 
 import gradlebuild.cleanup.extension.CleanupExtension
+import gradlebuild.cleanup.services.CachesCleaner
 import gradlebuild.integrationtests.tasks.DistributionTest
 
 plugins {
     java
+    id("gradlebuild.module-identity")
+}
+
+val intTestHomeDir = rootProject.layout.projectDirectory.dir("intTestHomeDir")
+
+val cachesCleanerService = gradle.sharedServices.registerIfAbsent("cachesCleaner", CachesCleaner::class) {
+    parameters.homeDir.set(intTestHomeDir)
 }
 
 tasks.withType<DistributionTest>().configureEach {
-    dependsOn(":cleanUpCaches")
-    finalizedBy(":cleanUpDaemons")
+    doFirst {
+        cachesCleanerService.get().cleanUpCaches(moduleIdentity.version.get())
+    }
```

Comment:
I kept the behavior the same as if it was a common task dependency. If we move the cleanup to the end - to an `AutoCloseable` build service, there arise race conditions with the daemon cleanup service - we'd probably have to merge them together and kill the daemons first (wait for them to really be killed on Windows) and only then clean the caches which might otherwise still be locked by stale processes.